### DISCUSSION TITLE
Issue retrieving variants by ID

### DISCUSSION THREAD

**Discussion by turismoracing** [swellstores/community#463] [Help] [answered]
Hello to everyone, this is my first post on this community.

I'm working on Plasmic creating all the frontend, and the time has come for me to retrieve information about my products on Swell.

I'm using cURL to get information about some variants but I'm unable to do it.

This is the query I'm using:

https://store-id:private-key@api.swell.store/products:variants/variant-id

When I retrieve the parent product I can get all the information normally, but the result for this one is the following:

![response](https://github.com/swellstores/community/assets/115661234/f4c7c007-ae6e-46c7-aeab-79e55ab79ab9)

Anyone can tell me what I'm doing wrong?

**Comment by logeshswell** [comment]
Can you share the exact URL information without private key?

I would also recommend you to fetch /products with an additional param: `expand: ["variants"]`

**Reply by logeshswell** [reply, marked as answer]
Another possible way could be,
https://store-id:private-key@api.swell.store/products:variants?parent_id=product-id
